lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -416,58 +409,112 @@ ioToListenError hostPreference portOpt e
                                    Core API
 -------------------------------------------------------------------------------}
 
-coreApiServer
-    :: forall ctx s t k n.
-        ( DelegationAddress n k
-        , Buildable (ErrValidateSelection t)
-        , k ~ ShelleyKey
-        , s ~ SeqState n k
-        , ctx ~ ApiLayer s t k
-        )
-    => ctx
-    -> Server (CoreApi n)
-coreApiServer ctx =
-    addresses ctx
-    :<|> wallets ctx
-    :<|> coinSelections ctx
-    :<|> transactions ctx
-    :<|> network ctx
-
-stakePoolServer
-    :: forall ctx s t n k.
-        ( DelegationAddress n k
+-- | A Servant server for our wallet API
+server
+    :: forall t n byron shelley.
+        ( shelley ~ ApiLayer (SeqState n ShelleyKey) t ShelleyKey
+        , byron ~ ApiLayer (RndState 'Mainnet) t ByronKey
+        -- yoroi ~ ApiLayer (SeqState n ByronKey) t ByronKey
```

Comment:
[question] What's the reason for putting these types in the constraints, rather than directly, or as type aliases? e.g.

```haskell
type ShelleyApiLayer (n :: NetworkDiscriminant) t = ApiLayer (SeqState n ShelleyKey) t ShelleyKey
```

Reply:
Actually no particular reason. But such type alias may be justified indeed :+1: